### PR TITLE
[DYN-3486] Extensions menu

### PR DESCRIPTION
**Pull request by Astul-Betizagasti**
### Purpose

Add a new Extensions menu which will include a menu item for each extension added in alphabetical order

Picture of how it looks with Dynamo's default extensions:

If for some reason no extensions are added then the menu will be disabled:

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

Aaron Tang (@QilongTang )

### FYIs

Roberto Tellez (@RobertGlobant20 )

### CODE REVIEW

**Comment by QilongTang** [comment]
Test Result (3 failures / +3)
DynamoCoreWpfTests.DocumentationBrowserViewExtensionTests.ClickingMenuItemLaunchesSidebarWithIndexContent
DynamoCoreWpfTests.DocumentationBrowserViewExtensionTests.DocsExtensionAddsMenuItem
DynamoCoreWpfTests.DocumentationBrowserViewExtensionTests.ShowingStartPageHidesBrowser

**Comment by Astul-Betizagasti** [comment]
> Test Result (3 failures / +3)
> DynamoCoreWpfTests.DocumentationBrowserViewExtensionTests.ClickingMenuItemLaunchesSidebarWithIndexContent
> DynamoCoreWpfTests.DocumentationBrowserViewExtensionTests.DocsExtensionAddsMenuItem
> DynamoCoreWpfTests.DocumentationBrowserViewExtensionTests.ShowingStartPageHidesBrowser

@QilongTang, I just updated the pr with a fix for these

File: src/DynamoCoreWpf/Extensions/MenuBarTypeExtensions.cs [hunk 1]
```diff
@@ -23,6 +23,7 @@ internal static string ToDisplayString(this MenuBarType type)
                 case MenuBarType.View: return Properties.Resources.DynamoViewViewMenu;
                 case MenuBarType.Help: return Properties.Resources.DynamoViewHelpMenu;
                 case MenuBarType.Packages: return Properties.Resources.DynamoViewPackageMenu;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Extensions/MenuBarTypeExtensions.cs:25]: Can you add a TODO to the cases above mentioning these will be deleted in Dynamo 3.0?

File: src/DynamoCoreWpf/Extensions/ViewLoadedParams.cs [hunk 2]
```diff
@@ -204,7 +224,8 @@ public enum MenuBarType
         Edit,
         View,
         Help,
-        Packages
+        Packages,
+        Extensions
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Extensions/ViewLoadedParams.cs:228]: Can you add a TODO to the cases above mentioning these will be deleted in Dynamo 3.0?

**Comment by Astul-Betizagasti** [comment]
@QilongTang I added the TODO's as you asked and also modify my implementation of the new api so that it doesn't use the code that is planed to be removed so there will be less changes needed in the future. Let me know if you have any further comments, thanks!

File: src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.cs [hunk 3]
```diff
@@ -70,7 +70,7 @@ public override void Loaded(ViewLoadedParams viewLoadedParams)
             this.documentationBrowserMenuItem = new MenuItem { Header = Resources.MenuItemText, IsCheckable = true };
             this.documentationBrowserMenuItem.Checked += MenuItemCheckHandler;
             this.documentationBrowserMenuItem.Unchecked += MenuItemUnCheckedHandler;
-            this.viewLoadedParamsReference.AddMenuItem(MenuBarType.View, this.documentationBrowserMenuItem);
+            this.viewLoadedParamsReference.AddExtensionMenuItem(this.documentationBrowserMenuItem);
```
Comment (QilongTang) [inline src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.cs:73]: Thanks, I think this makes most sense given none of the other MenuBarTypes will be supported for the first version of this API, if we plan to support them in the future, we can add another version of this API.

**Review by QilongTang — APPROVED** [review]
LGTM

**Pull request merged**